### PR TITLE
more useful vela system dry-run

### PR DESCRIPTION
**Pull request by wonderflow**
1. Add a demo for `vela system dry-run` which is handy for `debug`
2. fix a bug of `CreateOrUpdateObjects` that only update the first object.

### CODE REVIEW

File: pkg/utils/common/common.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ import (
 	"cuelang.org/go/cue"
 	"cuelang.org/go/encoding/openapi"
 	"github.com/AlecAivazis/survey/v2"
+	"github.com/ghodss/yaml"
```
Comment (ryanzhang-oss) [inline pkg/utils/common/common.go:19]: is this the right yaml package?

File: references/cli/dryrun.go [hunk 2]
```diff
@@ -1,26 +1,37 @@
...
 import (
+	"bytes"
 	"context"
 	"encoding/json"
+	"fmt"
 	"io/ioutil"
+	"os"
 	"path/filepath"
 
+	"k8s.io/apimachinery/pkg/runtime"
```
Comment (ryanzhang-oss) [inline references/cli/dryrun.go:12]: please reorg goimports

**Review by ryanzhang-oss — APPROVED** [review]

File: pkg/utils/common/common.go (hunk 1 above)
Comment (wonderflow) [reply pkg/utils/common/common.go:19]: yes

**Comment by ryanzhang-oss** [comment]
btw,  do we have any examples using the "kube" package in cue?

**Comment by wonderflow** [comment]
@ryanzhang-oss  Will improve it in another PR. Working for it now. For quick glance:

```
apiVersion: core.oam.dev/v1alpha2
kind: WorkloadDefinition
metadata:
  name: myworker
spec:
  definitionRef:
    name: deployments.apps
  schematic:
    cue:
      template: |
        import ("kube")
        output: kube.#Deployment

        output: {
        	spec: {
        		selector: matchLabels: {
        			"app.oam.dev/component": context.name
        		}

        		template: {
        			metadata: labels: {
        				"app.oam.dev/component": context.name
        			}

        			spec: {
        				containers: [{
        					name:  context.name
        					image: parameter.image

        					if parameter["cmd"] != _|_ {
        						command: parameter.cmd
        					}
        				}]
        			}
        		}
        	}
        }

        parameter: {
        	// +usage=Which image would you like to use for your service
        	// +short=i
        	image: string
        	// +usage=Commands to run in the container
        	cmd?: [...string]
        }
```

**Pull request merged**